File: week2/scrabble/scrabble.c
#include <ctype.h>
#include <stdio.h>
#include <string.h>

// Points assigned to each letter of the alphabet
int POINTS[] = {1, 3, 3, 2, 1, 4, 2, 4, 1, 8, 5, 1, 3, 1, 1, 3, 10, 1, 1, 1, 1, 4, 4, 8, 4, 10};

int compute_score(char *word)
{
    int score = 0;
    for (int i = 0; i < strlen(word); i++)
    {
        if (isupper(word[i]))
        {
            score += POINTS[word[i] - 'A'];
        }

        else if (islower(word[i]))
        {
            score += POINTS[word[i] - 'a'];
        }

    }
    return score;
}

int main(void)
{
    // Get input words from both players
    char word1[200];
    char word2[200];
    printf("Player 1: ");
    scanf("%s", word1);
    printf("Player 2: ");
    scanf("%s", word2);

    // Score both words
    int score1 = compute_score(word1);
    int score2 = compute_score(word2);

    // TODO: Print the winner
    if (score1 == score2)
    {
        printf("Tie!\n");
    }
    else if (score1 > score2)
    {
        printf("Player 1 wins!\n");
    }
    else
    {
        printf("Player 2 wins!\n");
    }
}

File: week2/scrabble/scrabble.py
def main():
    l = 'A B C D E F G H I J K L M N O P  Q R S T U V W X Y  Z'.split()
    p = '1 3 3 2 1 4 2 4 1 8 5 1 3 1 1 3 10 1 1 1 1 4 4 8 4 10'.split()
    points = {l: int(p) for l, p in zip(l, p)} | {l.lower(): int(p) for l, p in zip(l, p)}
    scores = []
    i = 1
    while i < 3:
        tmp = input(f'Player {i}: ').strip(' -_=+[{]}\|;:",<.>/?!@#$%^&*()1234567890~`')
        try:
            scores.append(sum([points[c] for c in tmp]))
        except KeyError:
            continue
        i += 1
    if scores[0] == scores[1]:
        print('Tie!')
    else:
        print(f'{"Player 1" if scores[0] > scores[1] else "Player 2"} wins!')


main()
